app/controllers/api/v1/soap_api_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Api::V1::SoapApiController < Api::V1::BaseController
+  soap_service namespace: 'urn:WashOut', wsse_auth_callback: ->(email, password) {
+    user = User.find_by(email: email)
+    return false unless user.present?
+    return user.valid_password?(password)
+  }
+
+  soap_action :get_final_cites_certificate,
+              args: {
+                CertificateNumber: :string,
+                TokenId: :string,
+                IsoCountryCode: :string
+              },
+              return: :string
+  def get_final_cites_certificate
+    render xml: Adapters::SimpleAdapter.run(Adapter.first).to_xml
```

Comment:
I've added a before_action that loads the adapter based on the provided country iso code. It also raises a Soap Fault if the adapter is not found.
Thank you! :)